examples/keyboard_layout.rs

```diff
@@ -898,6 +898,37 @@ fn with_daemon<F: Fn(Box<dyn Daemon>)>(f: F) {
     f(Box::new(server));
 }
 
+#[cfg(target_os = "macos")]
+fn macos_init() {
+    // This command returns Dark if we should use the dark theme
+    // defaults read -g AppleInterfaceStyle
+    if let Ok(output) = process::Command::new("defaults")
+        .arg("read")
+        .arg("-g")
+        .arg("AppleInterfaceStyle")
+        .output()
+    {
+        if output.stdout.starts_with(b"Dark") {
+            let _ = env::set_var("GTK_THEME", "Adwaita:dark");
```

Comment:
I might hope there's a way for a program to set it's theme other than an environmental variable, but I don't see one. I guess GTK isn't designed for an application to set that itself.